src/components/Deck.test.js

```diff
@@ -0,0 +1,61 @@
+import { DeckTrick } from './Deck'
+
+const deck = [
+    [{	suit: "spades", number: "Q"},
+     {	suit: "clubs", number: "4"},
+     {	suit: "hearts", number: "5"},
+     {	suit: "clubs", number: "3"},
+     {	suit: "clubs", number: "5"},
+     {	suit: "diamonds", number: "4"},
+     {	suit: "diamonds", number: "5"}],
+
+    [{	suit: "diamonds", number: "K"},
+     {	suit: "hearts", number: "10"},
+     {	suit: "spades", number: "10"},
+     {	suit: "hearts", number: "9"},
+     {	suit: "diamonds", number: "2"},
+     {	suit: "spades", number: "J"},
+     {	suit: "spades", number: "3"}],
+
+    [{	suit: "clubs", number: "8"},
+     {	suit: "spades", number: "5"},
+     {	suit: "hearts", number: "J"},
+     {	suit: "spades", number: "9"},
+     {	suit: "clubs", number: "9"},
+     {	suit: "spades", number: "A"},
+     {	suit: "clubs", number: "A"}]
+];
+
+const deckAfter = [
+    [{	suit: "spades", number: "Q"},
+     {	suit: "clubs", number: "3"},
+     {	suit: "diamonds", number: "5"},
+     {	suit: "hearts", number: "J"},
+     {	suit: "spades", number: "A"},
+     {	suit: "hearts", number: "10"},
+     {	suit: "diamonds", number: "2"}],
+
+    [{	suit: "clubs", number: "4"},
+     {	suit: "clubs", number: "5"},
+     {	suit: "clubs", number: "8"},
+     {	suit: "spades", number: "9"},
+     {	suit: "clubs", number: "A"},
+     {	suit: "spades", number: "10"},
+     {	suit: "spades", number: "J"}],
+
+    [{	suit: "hearts", number: "5"},
+     {	suit: "diamonds", number: "4"},
+     {	suit: "spades", number: "5"},
+     {	suit: "clubs", number: "9"},
+     {	suit: "diamonds", number: "K"},
+     {	suit: "hearts", number: "9"},
+     {	suit: "spades", number: "3"}]
+];
+
+const row = 2;
+
+describe("DeckTrick", () => {
+    it("simulates the 11th card trick on an array of three arrays", () => {
+ 	expect(DeckTrick({'rows': deck}, row)).toEqual(deckAfter);
```

Comment:
é interessante que o lado direito da comparação `toEqual` esteja escrito literalmente dentro do `toEqual`. O lado direito você pode organizar dentro do contexto, de preferência dentro do describe mesmo, não fora.

A lógica em cima disso é a seguinte, os contextos (describe) descrevem "o quê" você tá testando, e pra cada "o quê" você pode verificar várias coisas em `it` diferentes. Já o que está do lado direito do `toEqual` é o que você espera, isso é particular de cada `it` então merece ficar ali mesmo. Note que não há ganho nem perda de quantidade de código, mas você tem uma leitura do que você espera muito mais direta.